### ISSUE TITLE
Meta is always fetch from database

### ISSUE THREAD

**Issue by devfelipereis** [kodeine/laravel-meta#17]
Hello!

For example: I have a list of 10 movies and these movies have some metas. So, when a get all these movies, I have notice that 10 queries where executed to get all metas for each movie. 
In a situation where I don't need these metas, I have this queries in the same way.

So, is there any way to not get all that data automatically? Cause I have a situation where I count some data and I don't need any meta. I'm thinking about it cause for each new data in the database, a new query will be executed later. See the image below:

![log](https://cloud.githubusercontent.com/assets/5348752/11982099/8b61ec92-a98f-11e5-86eb-76a5bed3a4e2.png)

Any help? Thanks!

**Comment by kmccarth** [comment]
@devfelipereis I am having the same issue. I tried adding "meta" as a hidden field to the main model, but that didn't work.

**Comment by kmccarth** [comment]
@devfelipereis quick update...I didn't want the meta_data attribute returning at all, so I copy-and-pasted this function from Metable (https://github.com/kodeine/laravel-meta/blob/master/src/Kodeine/Metable/Metable.php#L252) and dropped it in my model. 

Instead of that function, my model contains a function like this:`  public function toArray()
  {
      return parent::toArray();
  }`

Obviously, there's some conditional logic you could to do get different meta data back based on the return, but I figured I'd drop you a note.

**Comment by devfelipereis** [comment]
@kmccarth I'll try this later and also try a conditional to get all metas when I want it. Thanks for your help.

**Comment by kodeine** [comment]
@devfelipereis i will think on how we can implement it in a package as well.

**Comment by dneykov** [comment]
Any plans for this implementation soon?
Thanks.

**Comment by kodeine** [comment]
@devfelipereis i'll try to work on it next week. It should be easier, just need to add Cache::Remember in few places.

**Comment by dneykov** [comment]
Still waiting on that update. Any progress? Thanks.

**Comment by dhcmega** [comment]
Hi, has this ever been implemented? is there a solution for this situation? thanks!
maybe a "withoutMetas" method?

**Comment by ERROPiX** [comment]
@dhcmega @dneykov I have just tested this package today, and I confirm it doesn't query for meta data if you don't use any, but as soon as you request a meta it will load them all in one query, so it's already done

The issue I discovered is is that it query meta table like this
`select * from `users_meta` where `user_id` = '1' and `user_id` is not null`
while I think it should do just like this:
`select * from `users_meta` where `user_id` = '1'`

**Comment by kodeine** [comment]
@ERROPiX can you share screenshot of sql?

**Comment by ERROPiX** [comment]
Of course, but it adds nothing to what I said in the text anyway:
![metable](https://user-images.githubusercontent.com/6576868/38056070-4b65c3d2-32d3-11e8-8956-53a185e4170c.png)
FYI, I am using Laravel 5.6

**Comment by dhcmega** [comment]
@ERROPiX the "is not null" is added by laravel, there is a bug related: https://github.com/laravel/framework/commit/1e80162e9a3a4c42a0079bd9f34e785197a66c07

**Comment by dhcmega** [comment]
The problem is here:

```
    public function toArray()
    {
        return array_merge(parent::toArray(), [
            'meta_data' => $this->getMeta()->toArray(),
        ]);
    }
```
it could be something like:

```
    public function toArray($withMeta = true)
    {
        if ($withMeta) {
            return array_merge(parent::toArray(), [
                'meta_data' => $this->getMeta()->toArray(),
            ]);
        }
        
        return parent::toArray();
    }
```

**Comment by dhcmega** [comment]
Hi!

I think this is the best way because when you are chaining you call toArray at the end, not all models will get the parameter.

```
    /**
     * Convert the model instance to an array.
     *
     * @return array
     */
    public function toArray()
    {
        return $this->hideMeta ?
            parent::toArray() :
            array_merge(parent::toArray(), [
                'meta_data' => $this->getMeta()->toArray(),
            ]);
    }
```

**Comment by kodeine** [comment]
@dhcmega can you please send a PR for proposed changes?

**Comment by dhcmega** [comment]
HI @kodeine 
Yes, I will test it this week and upload it.
Will be my first PR (letting you know, just in case)

**Comment by touqeershafi** [comment]
I also ran into the same problem? What is the solution, for now, to not call the meta query automatically?

**Comment by dhcmega** [comment]
@touqeershafi 
from the docs:

Prevent metas attribute from being populated
When you convert a model to an array (or json) and you don't need all meta fields, you can create a model's property to prevent metas from being added to the resulting array. You can also use it on eloquent relations.

```php
/* Post model */
public $hideMeta = true; // Do not add metas to array
```
